src/runtime/hexagon_dma.cpp

```diff
@@ -121,6 +127,150 @@ static void desc_pool_free (void* user_context) {
+    stDmaTransferParm.u16FrameStride        = dev->frame_stride;
+    stDmaTransferParm.u16RoiW               = roi_width;
+    stDmaTransferParm.u16RoiH               = roi_height;
+    stDmaTransferParm.u16RoiStride          = roi_stride;
+    stDmaTransferParm.bIsFmtUbwc            = dev->is_ubwc;
+    stDmaTransferParm.bUse16BitPaddingInL2  = 0;
+    stDmaTransferParm.pDescBuf              = desc_addr;
+    stDmaTransferParm.pTcmDataBuf           = dev->cache_buf;
+    stDmaTransferParm.pFrameBuf             = dev->buffer;
+    if (dev->is_write) {
+       stDmaTransferParm.eTransferType      = eDmaWrapper_L2ToDdr;
+    } else {
+       stDmaTransferParm.eTransferType      = eDmaWrapper_DdrToL2;
+    }
+    stDmaTransferParm.u16RoiX               = dev->offset_x + dst->dim[0].min;
+    stDmaTransferParm.u16RoiY               = dev->offset_y + dst->dim[1].min;
+
+    // DMA Driver Halves the Height and Y Offset so that only Half the ROI Size of Luma is transferred  for chroma
+    // We are compensating this assumption to meet  Halide perspective. i.e. ROI size is same for both Luma and Chroma    
+    if ((currentFmt == eDmaFmt_NV12_UV) ||
+        (currentFmt == eDmaFmt_P010_UV) ||
+        (currentFmt == eDmaFmt_TP10_UV) ||
+        (currentFmt == eDmaFmt_NV124R_UV)) {
+        stDmaTransferParm.u16RoiH = roi_height * 2;
+        stDmaTransferParm.u16RoiY = stDmaTransferParm.u16RoiY * 2;
+    }
+
+    debug(user_context) << "Hexagon: " << dev->dma_engine << " transfer: " << stDmaTransferParm.pDescBuf << "\n" ;
+    nRet = nDmaWrapper_DmaTransferSetup(dev->dma_engine, &stDmaTransferParm);
+    if (nRet != QURT_EOK) {
+        debug(user_context) << "Hexagon: DMA Transfer Error: " << nRet << "\n";
+        return halide_error_code_device_buffer_copy_failed;
+    }
+    
+    debug(user_context) << "Hexagon: " << dev->dma_engine << " move\n" ;
+
+    nRet = nDmaWrapper_Move(dev->dma_engine);
+    if (nRet != QURT_EOK) {
+        debug(user_context) << "Hexagon: nDmaWrapper_Move error: " << nRet << "\n";
+        return halide_error_code_device_buffer_copy_failed;
+    }
+    
+   debug(user_context) << "Hexagon: " << dev->dma_engine << " wait\n" ;
+
+    nRet = nDmaWrapper_Wait(dev->dma_engine);
+    if (nRet != QURT_EOK) {
+        debug(user_context) << "Hexagon: nDmaWrapper_Wait error: " << nRet << "\n";
+        return halide_error_code_device_buffer_copy_failed;
+    }
+
+    uint8_t *dest = reinterpret_cast<uint8_t *>(dst->host);
+
+    // Copy from Locked Cache to a temp DDR buffer
+    // TODO: This should be removed once the cache locking is addressed inside Halide Pipeline
+    if (dest) {
+        const int pixelsize = dst->type.bytes();
+        const int linesize = roi_width * pixelsize;
+        uint8_t *cache_buf = reinterpret_cast<uint8_t *>(dev->cache_buf);
+        debug(user_context) << "copy cache: " << "pixelsize= " << pixelsize << " linesize= " << linesize << "\n";
+        int y, sy, dy;
```

Comment:
I'd prefer it if these were declared "inline":

        for (int y = 0; y < roi_height; y++) {
            int sy = y * roi_stride * pixelsize;
            int dy = y * dst->dim[1].stride * pixelsize;
            memcpy(&dest[dy], &cache_buf[sy], linesize);
        }

The generated code should be the same, and it's easier to read/safer (IMO). Also note spacing changes.